Adding 8 FAQs to our docs (#6266)
harness/developer-hub: kb/platform/harness-platform-faqs.md

However, if you have legacy delegates in your first generation, you will require new delegate installations.

#### How do I support Docker in Docker for Harness Delegates?

To support Docker in Docker for Harness Delegates, follow these steps:

Check Kubernetes Cloud Provider Compatibility: Ensure that your Kubernetes Cloud Provider supports this configuration. For instance, note that Amazon EKS has ended support for Dockershim, which might affect this approach.

**Set up Delegate Image:**

- Utilize the INIT_SCRIPT variable or build a custom delegate image.
- Install necessary packages in the image. Here's an example of installations:

```
microdnf install yum
yum install -y yum-utils
yum-config-manager --add-repo https://download.docker.com/linux/centos/docker-ce.repo
yum install docker-ce docker-ce-cli containerd.io docker-buildx-plugin docker-compose-plugin
```

> **Update Delegate YAML:**

Reviewer: ```suggestion
**Update delegate YAML:**
```

- Add volume mounts to enable Docker functionality within the delegate container.

```yaml
volumes:
  - name: docker-sock
    hostPath:
      path: '/var/run/docker.sock'
  - name: docker-directory
    hostPath:
      path: '/var/lib/docker'
```

**Configure volume mounts in the YAML file:**

```yaml
volumeMounts:
  - mountPath: /var/run/docker.sock
    name: docker-sock
    readOnly: false
  - mountPath: '/var/lib/docker'
    name: docker-directory
    readOnly: false
```

> By following these steps, your delegate container should be able to execute Docker commands successfully by utilizing the `docker.sock` from your node hosts. This setup allows for Docker in Docker functionality within your Harness delegate container.

Reviewer: ```suggestion
By following these steps, your delegate container should be able to execute Docker commands successfully by utilizing the `docker.sock` from your node hosts. This setup allows for Docker in Docker functionality within your Harness Delegate container.
```

No, we don't support ldap secret engine.

#### Why am I experiencing errors with core_secret_access messages?

This error occurs when a user possesses the permission to execute a pipeline, yet lacks access to the project/organization/account secrets. This issue becomes critical when the pipeline's configuration includes resources that rely on these secrets. Consequently, users attempting to trigger or view the execution without proper secret access will encounter this error message.

> To fix it you just need to give the expected permissions to the User's group, for example, for secrets, there are View, Create/Edit, Delete, and Access permissions.

Reviewer: ```suggestion
To fix it, you must grant the expected permissions to the User's group, for example, for secrets, there are View, Create/Edit, Delete, and Access permissions.
```
